src/scripts/find-successful-workflow.js

```diff
@@ -92,9 +92,20 @@ function commitExists(commitSha) {
   }
 }
 
-async function isWorkflowSuccessful(pipelineId) {
-  return getJson(`https://circleci.com/api/v2/pipeline/${pipelineId}/workflow`)
-    .then(({ items }) => items.every(item => item.status === 'success'));
+async function isWorkflowSuccessful(pipelineId, workflowName) {
+  if (!workflowName || workflowName.length === 0) {
+    return getJson(`https://circleci.com/api/v2/pipeline/${pipelineId}/workflow`)
+      .then(({ items }) => items.every(item => item.status === 'success'));
+  } else {
+    return getJson(`https://circleci.com/api/v2/pipeline/${pipelineId}/workflow`).then(
+      function(items){
+        return items.items.some(
```

Comment:
Instead of `Array.some`, we should use `Array.find` first to find workflow based on name (each pipeline should have just a single workflow run) and then check for success. 

```JavaScript
return getJson(`https://circleci.com/api/v2/pipeline/${pipelineId}/workflow`).then(
  ({ items }) => {
     const workflowRun = items.find(item => item.name === workflowName);
     return workflowRun && workflowRun.status === 'success'
   });
```